rotkehlchen/tests/unit/decoders/test_rainbow.py

```diff
@@ -0,0 +1,222 @@
+from typing import Final
+
+import pytest
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.evm.decoding.constants import CPT_GAS
+from rotkehlchen.chain.evm.decoding.rainbow.constants import (
+    CPT_RAINBOW_SWAPS,
+    RAINBOW_ROUTER_CONTRACT,
+)
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.evm_event import EvmEvent
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.tests.utils.ethereum import get_decoded_events_of_transaction
+from rotkehlchen.types import Location, TimestampMS, deserialize_evm_tx_hash
+
+A_AIT: Final = Asset('eip155:1/erc20:0x89d584A1EDB3A70B3B07963F9A3eA5399E38b136')
+A_BUOY: Final = Asset('eip155:1/erc20:0x289Ff00235D2b98b0145ff5D4435d3e92f9540a6')
+A_AXGT: Final = Asset('eip155:1/erc20:0xDd66781D0E9a08D4FBb5eC7BAc80B691BE27F21D')
+A_ZIG: Final = Asset('eip155:1/erc20:0xb2617246d0c6c0087f18703d576831899ca94f01')
+
+
+@pytest.mark.parametrize('ethereum_accounts', [['0xfe25d6300D33e19E15AeeFEFD0Aafb319Dd61ae1']])
+def test_rainbow_swap_eth_to_token(ethereum_inquirer, ethereum_accounts):
```

Comment:
```suggestion
@pytest.mark.vcr(filter_query_parameters=['apikey'])
@pytest.mark.parametrize('ethereum_accounts', [['0xfe25d6300D33e19E15AeeFEFD0Aafb319Dd61ae1']])
def test_rainbow_swap_eth_to_token(ethereum_inquirer, ethereum_accounts):
```
You will need to VCR these tests to record the network calls and avoid making queries to etherscan here every time the CI runs. See https://docs.rotki.com/contribution-guides/python-testing.html#using-vcr

Feel free to ask questions if you need help with VCRing the tests.